Bearded.Utilities/Algorithms/BinPacking.cs

```diff
@@ -123,11 +123,10 @@ public static Result<T> Pack<T>(IEnumerable<Rectangle<T>> rectangles)
         /// If true, the algorithm is run for multiple heuristics and returns the best result.
         /// If false, it is only run for one heuristic, packing rectangles in order of decreasing area, which empirically works well.</param>
         /// <returns>
-        /// Null, if the given list of rectangles is empty.
-        /// Otherwise, an object containing the packed rectangles and some additional information.
+        /// A result object containing the packed rectangles and some additional information.
         /// </returns>
         /// <exception cref="ArgumentNullException"><paramref name="rectangles"/> is null.</exception>
-        public static Result<T> Pack<T>(IEnumerable<Rectangle<T>> rectangles, bool tryMultipleHeuristics)
+        public static Result<T>? Pack<T>(IEnumerable<Rectangle<T>> rectangles, bool tryMultipleHeuristics)
```

Comment:
Optional (existing issue):

Booleans in public method interfaces are always a bit of a code smell. The method exerts different behaviours based on the value, so shouldn't it just be two methods? Or is it having more than one responsibility?

Is there a way we can refactor this so that we avoid the boolean? Maybe a specific method for "PackWithMultipleHeuristics", or even a method where you pass in the different heuristics yourself.

Reply:
There are now separate methods.